kesconf/config.go

```diff
@@ -304,6 +304,20 @@ func ymlToServerConfig(y *ymlFile) (*File, error) {
 		if api.Timeout.Value < 0 {
 			return nil, fmt.Errorf("kesconf: invalid timeout '%d' for API '%s'", api.Timeout.Value, path)
 		}
+
+		// If mTLS authentication is disabled for at least one API,
+		// we must no longer require that a client sends a certificate.
+		// However, this may cause authentication errors when a client
+		// (the client's HTTP/TLS stack) does not send a certificate
+		// for an API that requires authentication.
+		if api.InsecureSkipAuth.Value {
+			if clientAuth == tls.RequireAnyClientCert {
+				clientAuth = tls.RequestClientCert
+			}
+			if clientAuth == tls.RequireAndVerifyClientCert {
+				clientAuth = tls.VerifyClientCertIfGiven
+			}
+		}
```

Comment:
@aead just to confirm, how exactly would someone disable auth for an API? I don't recall this in the server config options off the top of my head.

Reply:
https://github.com/minio/kes/blob/master/server-config.yaml#L59L101